### ISSUE TITLE
2.0.20-Beta2: IncompatibleClassChangeError

### ISSUE THREAD

**Issue by hfhbd** [google/ksp#1992]
I know, there is no 2.0.20-Beta2 compatible release. But using ksp 2.0.20-Beta1-1.0.22 with Kotlin 2.0.20-Beta2 results into this error:
```stacktrace
java.lang.IncompatibleClassChangeError: class com.google.devtools.ksp.common.PersistentMap cannot inherit from final class org.jetbrains.kotlin.com.intellij.util.io.PersistentHashMap
at com.google.devtools.ksp.common.IncrementalContextBase.<init>(IncrementalContextBase.kt:103)
        at com.google.devtools.ksp.IncrementalContext.<init>(IncrementalContext.kt:64)
        at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension$doAnalysis$2.invoke(KotlinSymbolProcessingExtension.kt:192)
        at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension$doAnalysis$2.invoke(KotlinSymbolProcessingExtension.kt:189)
        at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension.handleException(KotlinSymbolProcessingExtension.kt:414)
        at com.google.devtools.ksp.AbstractKotlinSymbolProcessingExtension.doAnalysis(KotlinSymbolProcessingExtension.kt:189)
        at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration(TopDownAnalyzerFacadeForJVM.kt:112)
```

Looking at the code, it looks like it is fixed by https://github.com/google/ksp/commit/bd04d69d8f79325453b5c702bbb769f9c0d4f418 but it is not yet released.

**Comment by InsanusMokrassar** [comment]
+1

**Comment by hfhbd** [comment]
Fixed with 1.0.23

**Comment by jchodev** [comment]
For me, it was error when I am using koltin version with "2.0.0"

![Image](https://github.com/user-attachments/assets/453eb30a-4f20-4b0c-8643-57c980ed9494)
